### DISCUSSION TITLE
Show a spinner/loader while hooks are running (?)

### DISCUSSION THREAD

**Discussion by valerio-pescatori** [extrawurst/gitui#2033] [Ideas]
Hi everyone! 

I recently moved from vscode to helix and I'm using gitui to manage the git part in the terminal as well and I'm loving it! It's really easy to use, it has all the functionalities I need and it's _really_ fast.

One thing I noticed though it's that if you have hooks setup, when you press enter to run the commit it kinda 'freezes'... obviously it's not freezing (bc it's **_blazing 💥 fast_**) it's just the hooks running in background.
But before realizing the first time it I was kinda confused and started pressing random keys to understand what's going on.
 
So I was thinking: is there a way to display a spinner or a loader somewhere just to tell the user that the commit is actually being done and it did not freeze? That would improve ux/dx a lot imo.

Thank to all the contributors!

**Comment by extrawurst** [comment]
see #2039 an issue created from this
